EasyImgur/FileHelper.cs

```diff
@@ -81,5 +83,50 @@ public static byte[] PeekBytes(this Stream stream, int count)
             stream.Seek(0, SeekOrigin.Begin);
             return buffer;
         }
+
+        public static bool CheckIfValidImageFileHeader(string path)
+        {
+            using (var fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, FileOptions.SequentialScan))
+            {
+                var bytes = fs.PeekBytes(4);
+                var imgFormatFromHeader = GetImageFormat(bytes);
+                return imgFormatFromHeader != null;
+            }
+        }
+
+        public static System.Drawing.Imaging.ImageFormat GetImageFormat(byte[] bytes)
+        {
+            // see http://www.mikekunz.com/image_file_header.html  
+            var bmp = Encoding.ASCII.GetBytes("BM");     // BMP
+            var gif = Encoding.ASCII.GetBytes("GIF");    // GIF
+            var png = new byte[] { 137, 80, 78, 71 };    // PNG
+            var tiff = new byte[] { 73, 73, 42 };         // TIFF
+            var tiff2 = new byte[] { 77, 77, 42 };         // TIFF
+            var jpeg = new byte[] { 255, 216, 255, 224 }; // jpeg
+            var jpeg2 = new byte[] { 255, 216, 255, 225 }; // jpeg canon
+
+            if (bmp.SequenceEqual(bytes.Take(bmp.Length)))
+                return ImageFormat.Bmp;
+
+            if (gif.SequenceEqual(bytes.Take(gif.Length)))
+                return ImageFormat.Gif;
+
+            if (png.SequenceEqual(bytes.Take(png.Length)))
+                return ImageFormat.Png;
+
+            if (tiff.SequenceEqual(bytes.Take(tiff.Length)))
+                return ImageFormat.Tiff;
+
+            if (tiff2.SequenceEqual(bytes.Take(tiff2.Length)))
+                return ImageFormat.Tiff;
+
+            if (jpeg.SequenceEqual(bytes.Take(jpeg.Length)))
+                return ImageFormat.Jpeg;
+
+            if (jpeg2.SequenceEqual(bytes.Take(jpeg2.Length)))
```

Comment:
There seems to be a lot of repetition here - perhaps the headers and their `ImageFormats` could be placed in an array of `KeyValuePairs`? I bet LINQ would work wonders with a `FirstOrDefault`.

Reply:
Certainly, this can be further improved.